### ISSUE TITLE
Provider produced inconsistent final plan

### ISSUE THREAD

**Issue by olksdr** [akamai/terraform-provider-akamai#88]
### Terraform Version
Terraform v0.12.18
+ provider.akamai v0.1.5

### Affected Resource(s)
Please list the resources as a list, for example:
- akamai_property_activation

### Terraform Configuration Files
```hcl
resource "akamai_property" "site" {
  name     = "site"
  contact  = ["user@example.com"]
  product  = "prd_Fresca"
  contract = data.akamai_contract.default.id
  group    = data.akamai_group.default.id
 
  hostnames = {
    for domain in local.domains : domain => akamai_edge_hostname.site[domain].edge_hostname
  }

  rule_format = "v2019-07-25"
  rules       = data.local_file.site_rules.content
  variables   = akamai_property_variables.site.json
}

resource "akamai_property_activation" "staging" {
  property = akamai_property.site.id
  version  = akamai_property.site.version
  network  = "STAGING"
  activate = true
  contact  = ["user@example.com"]
}
```

### Panic Output
```
Error: Provider produced inconsistent final plan

When expanding the plan for
akamai_property_activation.staging to include new
values learned so far during apply, provider "registry.terraform.io/-/akamai"
produced an invalid new value for .version: was cty.NumberIntVal(6), but now
cty.NumberIntVal(7).

This is a bug in the provider, which should be reported in the provider's own
issue tracker.

```

### Expected Behavior
The activation should be done on the first run without any errors

### Actual Behavior
The first `terraform apply` after changes produces error mentioned above.  The second run succeeds as expected. 

### Steps to Reproduce
1. Deploy the property and activate it on the `staging` network with `terraform apply`
3. Make the changes to the rules
4. Deploy the  new changed property and activate it on the `staging` network again with `terraform apply`

**Comment by ynohat** [comment]
@olksdr  I am running into this issue as well on TF 0.12.19 with Akamai provider 0.1.5; a workaround that may work for you: comment out the `version` field of the activation resource.

**Comment by olksdr** [comment]
@ynohat thanks for the suggestions. It will work for `staging` but my `production` relies on the property `version` and `stagning_version` and it should be set explicitly all the time. If the version is removed the `production` activation might pick up the latest property version, which might haven't been tested yet.  😄

**Comment by ynohat** [comment]
Yes, that is the tradeoff. YOLO driven development.

**Comment by sidph** [comment]
@olksdr @ynohat we are exploring how we solve for this quirky workflow. Our current thinking is to structure the activate resource such that:
1. Commenting out/not specifying the version just saves and activates the latest
2. providing the version, activate=true follows the current pattern. A first apply will save, following will activate
3. providing the version, setting activate to false will just keep saving

I suspect the above cty.NumberIntVal error comes down to the third case not being true today, so you enter some kind of a weird version mismatch state in your terraform state file. Thoughts?

**Comment by olksdr** [comment]
What I noticed with more testing today:
* I never hit this kind of error now while I'm modifying and activating only 1 property (using `terraform -target`)
* but when there is a bunch of the properties being activated (8 in my case) I hit this bug 100% of the time now. 

The workflow is simple: modify rules, run `terraform apply` and all the properties are modified as expected, rules deployed but activations are failing immediately with the above mentioned error. 


It looks to me that there are some race conditions in this case and the fetched versions do not match local state as you point @sidph

**Comment by olksdr** [comment]
@sidph  the third case you mentioned: I never used the activation property with `activate` set to `false` and can't report on this

**Comment by ynohat** [comment]
Hey @sidph I like the fact that an activation is just a resource.

Why can't we have:

* activate renamed to active (to match the declarative semantics, but that's sugar)
* active = true : activate if not already active
* active = false : deactivate if active

The use case for just managing the version and not deploying it is then handled by not including an activation resource.

Not sure this is related, but I also wonder if it is correct for the akamai_property resource to have production_version and staging_version as returned attributes, since that may be creating circular dependencies between the state of a version and the state of the activations. I wonder if it wouldn't be better to have those as data sources.

**Comment by sidph** [comment]
@ynohat primarily because we don't recommend deactivations. We are setup differently to other cloud providers - so the the recommendation is a create/destroy cycle means simply translates to a new version in Akamai. The above use case also doesn't account for users who want to save but not deploy

**Comment by ynohat** [comment]
@sidph The UI allows us to deactivate a version, not sure why terraform shouldn't allow it?

Saving and not deploying could be handled by not including an activation resource, or by including an activation resource that points to the currently active version (as retrieved using a data source that we would need to add, maybe)?

**Comment by IanCassTwo** [comment]
Surely the "Terraform way" of causing a managed resource to be destroyed would be to remove it from the .tf file? It doesn't make much sense to me to have a resource definition for a resource that doesn't exist (active = false : deactivate if active). I think the "activate = false" tag should simply cause the "apply" to do nothing - i.e. don't activate a new version but also don't deactivate existing.

**Comment by martinstibbe** [comment]
@olksdr This issue corrected in v0.2.0 also v0.3.0 coming out this next week has a fix for building 10 properties at same time I was able to d0 30 in sets of 10 with this library fix in place to lower traffic for contract group and cpcodes to help get past WAF exceptions
